### PR TITLE
add-jparepository-liquibase-ddlvalidate-softdelete-endpoints

### PR DESCRIPTION
**Pull request by andrey-kiyas**
Added:
- JpaRepository
- Liquibase
- DDLvalidate
- Softdelete
- Endpoints

### CODE REVIEW

File: src/main/java/com/onlinebookstore/exception/DataProcessingException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.onlinebookstore.exception;
+
+public class DataProcessingException extends RuntimeException {
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/exception/DataProcessingException.java]: what is it for? I do not see usage.

File: src/main/java/com/onlinebookstore/mapper/BookMapper.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.mapstruct.Mappings;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    @Mappings({
+            @Mapping(target = "id", ignore = true),
+            @Mapping(target = "deleted", ignore = true)
+    })
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/mapper/BookMapper.java]:
```suggestion
```

File: src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        return spec;
+    }
+}
+
+
+
+
+
+
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java]: empty lines

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/onlinebookstore/mapper/BookMapper.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/mapper/BookMapper.java]: Without this part I get a checkstyle error: "BookMapper.java:[17,10] Unmapped target properties: "id, deleted""

File: src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java (hunk 3 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java]: Yes, sorry, corrected

File: src/main/java/com/onlinebookstore/repository/book/spec/DescriptionSpecificationProvider.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.onlinebookstore.repository.book.spec;
+
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DescriptionSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/book/spec/DescriptionSpecificationProvider.java:10]: how it works? search by full Description?

File: src/main/java/com/onlinebookstore/repository/book/spec/PriceSpecificationProvider.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.onlinebookstore.repository.book.spec;
+
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/book/spec/PriceSpecificationProvider.java:10]: Same. We need price bounds for searchI think (min/max)

File: src/main/java/com/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "title";
+    }
+
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) -> root
+                .get("title")
+                .in(Arrays.stream(params).toArray());
```
Comment (Rommelua) [inline src/main/java/com/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java]: let's use "like" logic for Title
for example when we have "master" in parameter we would like to see both
Master and Margarita and Master Your Emotions in the results

File: src/main/java/com/onlinebookstore/mapper/BookMapper.java (hunk 2 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/mapper/BookMapper.java]: ...but checkstyle plugin 3.1.2 solved this problem. Corrected.

File: src/main/java/com/onlinebookstore/exception/DataProcessingException.java (hunk 1 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/exception/DataProcessingException.java]: Corrected, thanks!

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/onlinebookstore/repository/book/spec/PriceSpecificationProvider.java (hunk 5 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/book/spec/PriceSpecificationProvider.java:10]: Ok, added

File: src/main/java/com/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java (hunk 6 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/book/spec/TitleSpecificationProvider.java]: Done

File: src/main/java/com/onlinebookstore/repository/book/spec/DescriptionSpecificationProvider.java (hunk 4 above)
Comment (andrey-kiyas) [reply src/main/java/com/onlinebookstore/repository/book/spec/DescriptionSpecificationProvider.java:10]: Yes, after last correction

Comment (Rommelua) [reply src/main/java/com/onlinebookstore/repository/book/spec/DescriptionSpecificationProvider.java:10]: I think this class can be deleted at all. I've never seen a search by Description. But as you wish

**Review by Rommelua — APPROVED** [review]

**Pull request merged**